File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/decorator/solution/with_milk.cpp
// with_milk.cpp

#include <iostream>
#include "beverage.h"
#include "with_milk.h"

with_milk::with_milk(beverage * a_successor)
: with(a_successor)
{
}

void with_milk::prepare()
{
	with::prepare();
	std::cout << "Adding milk.\n";
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/decorator/solution/WithMilk.java
// WithMilk.java

public class WithMilk extends With
{
	public WithMilk(Beverage aSuccessor)
	{
		super(aSuccessor);
	}

	public void prepare()
	{
		super.prepare();
		System.out.println("Adding milk.");
	}
}
